functions.cpp

```diff
@@ -87,17 +87,24 @@ packToken Function::call(packToken _this, const Function* func,
 }
 
 /* * * * * class CppFunction * * * * */
+CppFunction::CppFunction() {
+    this->_name = "";
+    this->isStdFunc = false;
+    this->func = NULL;
+}
```

Comment:
Is this constructor really any useful? If someone tries to call it it might cause a segmentation fault or I am being paranoid?

Reply:
In some cases,an error accurred because of the cppfunction doesn't has a default constructor.So I add it to the codes.

Reply:
Hello @ZipoChan, one of the things I want for this library if for it to protect our users from unexpected and hard-to-track segmentation faults, and using this constructor might be a very easy way for the user of the library to cause one, here a sample code that could cause this problem:

```c++
#include <iostream>

#include"shunting-yard.h"

int main() {
  TokenMap tm = TokenMap();

  tm["inlineFunc"] = CppFunction();

  calculator c1("inlineFunc(10, 4)");

  std::cout << c1.eval(tm) << std::endl; // result should be 24

  return 0;
}
```

In the cases you mentioned that an error occurred I am imagining that these are static compilation errors, which are far easier to debug than segfaults. So I would prefer for the library to be a little harder to use than far harder to debug afterward.

Again I am sorry for taking so long to respond.